### PR TITLE
UtMethod removal

### PR DESCRIPTION
**Pull request by sergeypospelov**
# Description

- Substituted `UtMethod` with `ExecutableId`.
- Changed Kotlin Reflection to Java Reflection almost everywhere.
- Refactored util functions a bit. Now functions such as `Method.isStatic`, `Class.isAbstract` are located in the `ReflectionUtil.kt`.
- Cleaned up imports somewhere.
- Change some entry points to our test generation, more specifically, the actual code of receiving methods under test in CLI and Contest.
- Some local refactorings.

**Some context:**

`UtMethod` is a vestigial class which was used for passing method under test to our `TestCaseGenerator`. Now we use `ExecutableId` for identifying every method, so it's unnecessary to have one more entity. We can simply build `ExecutableId` from `java.reflect.Method` and `KCallable<*>` by calling `.executableId` extension function, assuming we have `UtContext` initialized.

`ExecutableId` has two main inheritors: `MethodId` and `ConstructorId`. We also have our own wrappers `ClassId` and `FieldId` for java classes and fields respectively.

Please note, that some reflection dependent parts of `FieldId`, `ClassId` and `ExecutableId` API, e.g. `MethodId.method`, `ClassId.isPublic`, etc., require `UtContext` to be set. It's very implicit and to be refactored in the next PRs.

## Type of Change

- Refactoring (typos and non-functional changes)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

## Manual Scenario 

- [x] Run test generation with Intellij plugin (nested and anonymous classes)

# Checklist:

- [x] The change followed the style guidelines of the UTBot project
- [x] Self-review of the code is passed
- [x] The change contains enough commentaries, particularly in hard-to-understand areas
- [x] New documentation is provided or existed one is altered
- [x] No new warnings
- [x] All tests pass locally with my changes

### CODE REVIEW

File: utbot-framework/src/main/kotlin/org/utbot/sarif/SarifReport.kt [hunk 1]
```diff
@@ -128,16 +128,16 @@ class SarifReport(
...
         uncheckedException: UtExecutionFailure
     ): Pair<SarifResult, SarifRule> {
 
         val exceptionName = uncheckedException.exception::class.java.simpleName
         val ruleId = "utbot.unchecked.$exceptionName"
 
-        val methodName = method.callable.name
-        val classFqn = method.clazz.qualifiedName
+        val methodName = method.name
+        val classFqn = method.classId.name
```
Comment (SBOne-Kenobi) [inline utbot-framework/src/main/kotlin/org/utbot/sarif/SarifReport.kt:140]: Introduce something like ExecutableId::fqnName

**Review by SBOne-Kenobi — CHANGES_REQUESTED** [review]
Good PR! There is some suggestions for a future replacement `SootMethod`: we can introduce `ReflectionWrapper` that will provide reflection methods and `ReflectionWrapperBuilder` that will create instances of `ReflectionWrappers` depending on specified implementation such as Soot or JCDB.

File: utbot-framework/src/main/kotlin/org/utbot/engine/Resolver.kt [hunk 2]
```diff
@@ -118,7 +118,7 @@ class Resolver(
     val typeRegistry: TypeRegistry,
     private val typeResolver: TypeResolver,
     val holder: UtSolverStatusSAT,
-    methodUnderTest: UtMethod<*>,
+    methodUnderTest: SootMethod,
```
[the remark this answers is not in the document]
Comment (sergeypospelov) [reply utbot-framework/src/main/kotlin/org/utbot/engine/Resolver.kt]: Fixed it.

File: utbot-framework/src/main/kotlin/org/utbot/engine/Traverser.kt [hunk 3]
```diff
@@ -1,3 +1,4 @@
+@file:Suppress("JAVA_MODULE_DOES_NOT_EXPORT_PACKAGE")
```
[the remark this answers is not in the document]
Comment (sergeypospelov) [reply utbot-framework/src/main/kotlin/org/utbot/engine/Traverser.kt:1]: My mistake, thank you.

File: utbot-framework/src/main/kotlin/org/utbot/engine/Traverser.kt [hunk 4]
```diff
@@ -200,7 +197,7 @@ import java.util.concurrent.atomic.AtomicInteger
 private val CAUGHT_EXCEPTION = LocalVariable("@caughtexception")
 
 class Traverser(
-    private val methodUnderTest: UtMethod<*>,
+    private val methodUnderTest: SootMethod,
```
[the remark this answers is not in the document]
Comment (sergeypospelov) [reply utbot-framework/src/main/kotlin/org/utbot/engine/Traverser.kt]: Fixed it

File: utbot-framework/src/main/kotlin/org/utbot/engine/UtBotSymbolicEngine.kt [hunk 5]
```diff
@@ -176,7 +177,7 @@ class UtBotSymbolicEngine(
     // TODO HACK violation of encapsulation
     internal val typeResolver: TypeResolver = TypeResolver(typeRegistry, hierarchy)
 
-    private val classUnderTest: ClassId = methodUnderTest.clazz.id
+    private val classUnderTest: ClassId = methodUnderTest.declaringClass.id
```
[the remark this answers is not in the document]
Comment (sergeypospelov) [reply utbot-framework/src/main/kotlin/org/utbot/engine/UtBotSymbolicEngine.kt]: Fixed it

File: utbot-framework/src/main/kotlin/org/utbot/engine/UtBotSymbolicEngine.kt [hunk 6]
```diff
@@ -423,9 +418,9 @@ class UtBotSymbolicEngine(
         }
 
         val methodUnderTestDescription = FuzzedMethodDescription(executableId, collectConstantsForFuzzer(graph)).apply {
-            compilableName = if (methodUnderTest.isMethod) executableId.name else null
-            className = executableId.classId.simpleName
-            packageName = executableId.classId.packageName
+            compilableName = if (!methodUnderTest.isConstructor) executableId.name else null
```
[the remark this answers is not in the document]
Comment (sergeypospelov) [reply utbot-framework/src/main/kotlin/org/utbot/engine/UtBotSymbolicEngine.kt]: Fixed it

File: utbot-framework/src/main/kotlin/org/utbot/sarif/SarifReport.kt (hunk 1 above)
Comment (sergeypospelov) [reply utbot-framework/src/main/kotlin/org/utbot/sarif/SarifReport.kt:140]: Looks like `KClass::qualifiedName` method returns the same thing as `classId.name`.

**Review by SBOne-Kenobi — APPROVED** [review]

**Review by IdeaSeeker — APPROVED** [review]

**Review by EgorkaKulikov — APPROVED** [review]
Comments discussed verbally

File: utbot-core/src/main/kotlin/org/utbot/common/ReflectionUtil.kt [hunk 7]
```diff
@@ -73,3 +74,41 @@ inline fun <reified R> Field.withAccessibility(block: () -> R): R {
...
+    get() = Modifier.isStatic(modifiers)
+
+val Member.isPrivate
+    get() = Modifier.isPrivate(modifiers)
+
+val Member.isPublic
+    get() = Modifier.isPublic(modifiers)
+
+val Member.isFinal: Boolean
+    get() = Modifier.isFinal(modifiers)
```
Comment (CaelmBleidd) [inline utbot-core/src/main/kotlin/org/utbot/common/ReflectionUtil.kt:93]: Please, make type declarations for functions consistent

File: utbot-framework-test/src/test/kotlin/org/utbot/framework/assemble/AssembleModelGeneratorTests.kt [hunk 8]
```diff
@@ -1444,9 +1446,9 @@ class AssembleModelGeneratorTests {
     private fun createModelsAndAssert(
         models: List<UtModel>,
         expectedModelRepresentations: List<String?>,
-        assembleTestUtils: UtMethod<*> = UtMethod.from(AssembleTestUtils::class.functions.first()),
+        assembleTestUtils: ExecutableId = AssembleTestUtils::class.id.allMethods.first(),
     ) {
-        val modelsMap = AssembleModelGenerator(assembleTestUtils).createAssembleModels(models)
+        val modelsMap = AssembleModelGenerator(assembleTestUtils.classId.packageName).createAssembleModels(models) // TODO: rewrite usages
```
Comment (CaelmBleidd) [inline utbot-framework-test/src/test/kotlin/org/utbot/framework/assemble/AssembleModelGeneratorTests.kt:1451]: Do we have a task for it?

File: utbot-framework/src/main/kotlin/org/utbot/engine/Traverser.kt [hunk 9]
```diff
@@ -957,23 +955,23 @@ class Traverser(
...
+        val callable = methodUnderTest.executable
         val type = if (identityRef is ThisRef) {
             // TODO: for ThisRef both methods don't return parameterized type
             if (methodUnderTest.isConstructor) {
-                methodUnderTest.javaConstructor?.annotatedReturnType?.type
+                callable.annotatedReturnType?.type
             } else {
-                callable.instanceParameter?.type?.javaType
-                    ?: error("No instanceParameter for ${callable.signature} found")
+                callable.declaringClass // same as it was, but it isn't parametrized type
```
Comment (CaelmBleidd) [inline utbot-framework/src/main/kotlin/org/utbot/engine/Traverser.kt:964]: What do you mean? We can't resolve generics for `this` instance anymore, can we?

File: utbot-framework/src/main/kotlin/org/utbot/tests/infrastructure/TestCodeGeneratorPipeline.kt [hunk 10]
```diff
@@ -231,7 +231,8 @@ class TestCodeGeneratorPipeline(private val testFrameworkConfiguration: TestFram
     private fun ClassPipeline.retrieveTestClassName(testSuffix: String): String =
         stageContext.classUnderTest.let { "${it.java.`package`.name}.${it.simpleName}" } + testSuffix
 
-    private fun List<KClass<*>>.createBuildDirectory() = FileUtil.isolateClassFiles(*toTypedArray()).toPath()
+    private fun List<KClass<*>>.createBuildDirectory() = // TODO: refactor KClass<*>
```
Comment (CaelmBleidd) [inline utbot-framework/src/main/kotlin/org/utbot/tests/infrastructure/TestCodeGeneratorPipeline.kt:234]: Refactor when?

File: utbot-core/src/main/kotlin/org/utbot/common/ReflectionUtil.kt (hunk 7 above)
Comment (sergeypospelov) [reply utbot-core/src/main/kotlin/org/utbot/common/ReflectionUtil.kt:93]: Missed it, thank you.

File: utbot-framework-test/src/test/kotlin/org/utbot/framework/assemble/AssembleModelGeneratorTests.kt (hunk 8 above)
Comment (sergeypospelov) [reply utbot-framework-test/src/test/kotlin/org/utbot/framework/assemble/AssembleModelGeneratorTests.kt:1451]: Actually, this is a useless TODO.

File: utbot-framework/src/main/kotlin/org/utbot/engine/Traverser.kt (hunk 9 above)
Comment (sergeypospelov) [reply utbot-framework/src/main/kotlin/org/utbot/engine/Traverser.kt:964]: I mean, that the previous code points to exactly the same object: they both point to declaringClass of type `java.lang.Class<*>`, which isn't a parametrized type

**Pull request merged**